nrf52840-hal/src/ieee802154.rs

```diff
@@ -226,6 +238,14 @@ impl<'c> Radio<'c> {
         self.needs_enable = true;
         match cca {
             Cca::CarrierSense => self.radio.ccactrl.write(|w| w.ccamode().carrier_mode()),
+            Cca::EnergyDetection {ed_threshold} => {
```

Comment:
LGTM

Reply:
> LGTM

my bad, insufficient lack of attention. this branch should do a single `write` operation. each `write` operation writes a new value to the entire register so this is doing:

``` rust
(0x4000166C as *mut u32).write_volatile(0); // CCAMODE = EdMode
(0x4000166C as *mut u32).write_volatile((ed_threshold as u32) << 8); // CCAEDTHRES = ed_threshold
```

This happens to work in this case because `ED_MODE = 0` but if you were configuring `Cca::CarrierAndEdMode` then doing it with 2 writes would result in setting the CCA mode to EdMode:

``` rust
// say we are configuring CcaMode::CarrierAndEdMode { ed_threshold }

// OK so far
(0x4000166C as *mut u32).write_volatile(2); // CCAMODE = CarrierAndEdMode

// !!! this second write sets the bits `0..2` to zero (and that's EdMode)
(0x4000166C as *mut u32).write_volatile((ed_threshold as u32) << 8); // CCAEDTHRES = ed_threshold
```

NB: `(0x4000166C as *mut u32).write_volatile(0xff)` is like `*(0x4000166C as *mut u32) = 0xff;` but has volatile semantics.